iris-mpc-cpu/src/hawkers/galois_store.rs

```diff
@@ -250,6 +250,9 @@ impl VectorStore for LocalNetAby3NgStoreProtocol {
         query: &Self::QueryRef,
         vectors: &[Self::VectorRef],
     ) -> Vec<Self::DistanceRef> {
+        if vectors.is_empty() {
+            return vec![];
+        }
```

Comment:
Do you think it might make sense to instead put this check in `eval_pairwise_distances` and `lift_distances`?  These functions are the lowest level where network operations are initiated based on the contents of a `Vec`, so putting the checks there future-proofs the design so any other function calls that need to do these computations also saves on the empty vector case automatically.

Reply:
The cloning here is quite heavy as we copy iris shares. Therefore, I left this one and added additional checks as per your suggestion.